ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/stats/HiveRelMdRowCount.java

```diff
@@ -170,7 +170,7 @@ public Double getRowCount(Sort rel, RelMetadataQuery mq) {
   @Override
   public Double getRowCount(Filter rel, RelMetadataQuery mq) {
     if (rel instanceof StatEnhancedHiveFilter) {
-      return rel.getRows();
+      return 1.0D;
```

Comment:
This seems wrong. It seems that we should rather do `((StatEnhancedHiveFilter) rel).getRowCount()`.